src/GraphPanel.java

```diff
@@ -3,20 +3,19 @@
 import java.awt.event.*;
 import java.awt.geom.GeneralPath;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import java.util.stream.Collectors;
 
 public class GraphPanel extends JPanel {
-    private double zoom = 1.0;
+    private final List<ParametricFunction> polynomials = new ArrayList<>();
```

Comment:
The name of this variable suggests there are only polynomials, however these could be parametric, too.

Reply:
Done